src/test/java/nearsoft/academy/bigdata/recommendation/MovieRecommender.java

```diff
@@ -0,0 +1,138 @@
+import java.nio.file.Paths;
+import java.nio.file.StandardOpenOption;
+import java.util.*;
+//import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import java.util.zip.GZIPInputStream;
+
+public class MovieRecommender{
+
+    int reviews = 0, userNum = 1, productNum = 1;
+    HashMap<String, Integer> users = new HashMap<String, Integer>();
+    HashMap<String, Integer> products = new HashMap<String, Integer>();
+
+
+public MovieRecommender(String file){
+    List<String> usersList = new ArrayList<String>();
+    List<String> productsList = new ArrayList<String>();
+    List<String> scoresList = new ArrayList<String>();
+
+    try{
+        InputStream fileStream = new FileInputStream(file);
+        InputStream gzipStream = new GZIPInputStream(fileStream);
+        Reader decoder = new InputStreamReader(gzipStream);
+        BufferedReader buffered = new BufferedReader(decoder);
+
+        PrintWriter pw = new PrintWriter(new File("movies.csv"));
+
+        String line;
+        StringBuilder sb = new StringBuilder();
+
+        while ((line = buffered.readLine()) != null){
+            if(line.contains("review/userId:")){
+                String[] parts = line.split("review/userId: ");
+                if(!users.containsKey(parts[1])) {
+                    users.put(parts[1], userNum);
+                    usersList.add("" + userNum);
+                    userNum++;
+                } else{
+                    usersList.add("" + users.get(parts[1]));
+                }
+            }else if(line.contains("product/productId:")){
+                reviews++;
+                String[] parts = line.split("product/productId: ");
+
+                if(!products.containsKey(parts[1])) {
+                    products.put(parts[1], productNum);
+                    productsList.add("" + productNum);
+                    productNum++;
+                } else{
+                    productsList.add("" + products.get(parts[1]));
+                }
+            }else if(line.contains("review/score:")){
+                String[] parts = line.split("review/score: ");
+                scoresList.add(parts[1]);
+            }
+        }
+
+        for (int counter = 0; counter < usersList.size(); counter++) {
+            pw.write(usersList.get(counter) + ',' + productsList.get(counter) + ',' + scoresList.get(counter) + '\n');
+            pw.println();
+        }
+
+        pw.close();
+
+        //pw.write(sb.toString());
+        //pw.close();
+    } catch(FileNotFoundException e) {
+        System.err.println("Caught FileNotFoundException: " + e.getMessage());
+    } catch(IOException e) {
+        System.err.println("Caught IOException: " + e.getMessage());
+    }
+}
+
+public int getTotalReviews(){
+    return reviews;
+}
+
+public int getTotalProducts(){
+    return products.size();
+}
+
+public int getTotalUsers(){
+    return users.size();
+}
+
+public <String> List<String> getRecommendationsForUser(String userID){
```

Comment:
I think that maybe the generic method here could be used as a normal method, because you will not pass parameters that inherit from String because this class is not inheritable :)